pkg/util/validation/validation.go

```diff
@@ -125,41 +125,29 @@ func ValidateSpreadConstraint(spreadConstraints []policyv1alpha1.SpreadConstrain
 }
 
 // ValidateOverrideSpec validates that the overrider specification is correctly defined.
-func ValidateOverrideSpec(overrideSpec *policyv1alpha1.OverrideSpec) error {
+func ValidateOverrideSpec(overrideSpec *policyv1alpha1.OverrideSpec) field.ErrorList {
+	var allErrs field.ErrorList
 	if overrideSpec == nil {
 		return nil
 	}
-
-	if overrideSpec.OverrideRules == nil {
-		return nil
+	specPath := field.NewPath("spec")
+	//nolint:staticcheck
+	// disable `deprecation` check for backward compatibility.
+	if overrideSpec.TargetCluster != nil {
+		allErrs = append(allErrs, ValidateClusterAffinity(overrideSpec.TargetCluster, specPath.Child("targetCluster"))...)
 	}
-
 	//nolint:staticcheck
 	// disable `deprecation` check for backward compatibility.
-	if overrideSpec.TargetCluster != nil || !EmptyOverrides(overrideSpec.Overriders) {
-		return fmt.Errorf("overrideRules and (overriders or targetCluster) can't co-exist")
+	if overrideSpec.TargetCluster != nil && overrideSpec.OverrideRules != nil {
+		allErrs = append(allErrs, field.Invalid(specPath.Child("targetCluster"), overrideSpec.TargetCluster, "overrideRules and targetCluster can't co-exist"))
 	}
-
-	for overrideRuleIndex, rule := range overrideSpec.OverrideRules {
-		rulePath := field.NewPath("spec").Child("overrideRules").Index(overrideRuleIndex)
-
-		// validates provided annotations.
-		for annotationIndex, annotation := range rule.Overriders.AnnotationsOverrider {
-			annotationPath := rulePath.Child("overriders").Child("annotationsOverrider").Index(annotationIndex)
-			if err := apivalidation.ValidateAnnotations(annotation.Value, annotationPath.Child("value")).ToAggregate(); err != nil {
-				return err
-			}
-		}
-
-		// validates provided labels.
-		for labelIndex, label := range rule.Overriders.LabelsOverrider {
-			labelPath := rulePath.Child("overriders").Child("labelsOverrider").Index(labelIndex)
-			if err := metav1validation.ValidateLabels(label.Value, labelPath.Child("value")).ToAggregate(); err != nil {
-				return err
-			}
-		}
+	//nolint:staticcheck
+	// disable `deprecation` check for backward compatibility.
+	if !EmptyOverrides(overrideSpec.Overriders) && overrideSpec.OverrideRules != nil {
```

Comment:
This function is only used in `pkg/util/validation/validation.go`, change name to `emptyOverrides`.

Reply:
Done.